src/exec/job.rs

```diff
@@ -53,7 +53,7 @@ pub fn batch(
     limit: usize,
 ) -> ExitCode {
     let paths = rx.iter().filter_map(|value| match value {
-        WorkerResult::Entry(path) => Some(path),
+        WorkerResult::Entry(val) => Some(val.path().to_owned()),
```

Comment:
Hm. Can we prevent the cloning here (`.to_owned()` essentially clones the input path)? We might be able to use `into_iter` instead of `iter`?